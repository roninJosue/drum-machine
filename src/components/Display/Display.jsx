import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components";
import "@fontsource/roboto"
import {primaryColor} from "../../utils/colors";


const DisplayStyled = styled.div`
  width: 200px;
  height: 56px;
  font-family: 'Roboto',monospace;
  padding: 1rem;
  border: 1px solid ${primaryColor};
  //text-transform: uppercase;
  display: flex;
  align-items: center;
`

const Display = ({text}) => {
  return (
    <DisplayStyled>
      {text}
    </DisplayStyled>
  );
};

Display.propTypes = {
  text: PropTypes.string.isRequired
};

export default Display;
