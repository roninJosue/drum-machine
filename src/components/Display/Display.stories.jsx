import React from "react"
import Display from "./Display";

export default {
  title: 'Display',
  component: Display
}

export const DisplayExample = () => {
  return <Display text={'Hello'} />
}