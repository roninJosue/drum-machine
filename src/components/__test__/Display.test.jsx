import React from "react"
import {render, screen} from "@testing-library/react";
import Display from "../Display";

describe('Display component', () => {

  it('Should render', async () => {
    const {asFragment} =  render(<Display text='hello'/>)

    const html = asFragment()

    expect(html).toMatchSnapshot()
  })

  it("Should contain the text 'OLLEH'",() => {
    render(<Display text='hello' />)
    screen.debug()

    const text = screen.getByText('OLLEH')

    expect(text).toHaveTextContent('OLLEH')
  });
})
