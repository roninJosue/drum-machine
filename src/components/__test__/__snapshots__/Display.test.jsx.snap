// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component Should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cPDXuc"
  >
    OLLEH
  </div>
</DocumentFragment>
`;
